src/repository/repository.go

```diff
@@ -1,200 +0,0 @@
-package repository
-
-import (
-	"archive/tar"
-	"compress/gzip"
-	"context"
-	"encoding/json"
-	"errors"
-	"io"
-	"net/url"
-	"os"
-	"path/filepath"
-
-	"github.com/blocklessnetworking/b7s/src/db"
-	"github.com/blocklessnetworking/b7s/src/http"
-	"github.com/blocklessnetworking/b7s/src/models"
-	log "github.com/sirupsen/logrus"
-)
-
-type JSONRepository struct{}
-
-// get the manifest from the repository
-// downloads the binary
-func (r JSONRepository) Get(ctx context.Context, installMsg models.MsgInstallFunction) (models.FunctionManifest, error) {
-	WorkSpaceRoot := ctx.Value("config").(models.Config).Node.WorkspaceRoot
-
-	functionManifest := models.FunctionManifest{}
-	err := http.GetJson(installMsg.ManifestUrl, &functionManifest)
-
-	if err != nil {
-		log.Warn(err)
-	}
-
-	if functionManifest.Runtime.Url != "" {
-		DeploymentUrl, err := url.Parse(functionManifest.Runtime.Url)
-		if err != nil {
-			log.Warn(err)
-		}
-
-		ManifestUrl, err := url.Parse(installMsg.ManifestUrl)
-		if err != nil {
-			log.Warn(err)
-		}
-
-		if DeploymentUrl.Host == "" {
-			DeploymentUrl.Host = ManifestUrl.Host
-		}
-
-		if DeploymentUrl.Scheme == "" {
-			DeploymentUrl.Scheme = ManifestUrl.Scheme
-		}
-
-		functionManifest.Deployment = models.Deployment{
-			Uri:      DeploymentUrl.String(),
-			Checksum: functionManifest.Runtime.Checksum,
-		}
-	}
-
-	cachedFunction, err := db.GetString(ctx, installMsg.Cid)
-	WorkSpaceDirectory := WorkSpaceRoot + "/" + installMsg.Cid
-
-	if err != nil {
-		if err.Error() == "pebble: not found" {
-			log.Info("function not found in cache, syncing")
-		} else {
-			log.Warn(err)
-		}
-	}
-
-	if cachedFunction == "" {
-		// download the function
-		fileName, err := http.Download(ctx, functionManifest)
-
-		UnGzip(fileName, WorkSpaceDirectory)
-		if err != nil {
-			log.Warn(err)
-		}
-
-		functionManifest.Deployment.File = fileName
```

Comment:
Example I mentioned - setting the deployment file path to the archive path, even though we potentially extracted a `.wasm` file elsewhere.

Reply:
let me get you more information around this